addon/components/step-manager.ts

```diff
@@ -95,12 +95,12 @@ export default class StepManagerComponent extends TaglessComponent {
     set(this, 'transitions', new StateMachine(initialStep));
   }
 
-  @computed('transitions.{currentStep,length}')
```

Comment:
Why change this computed property? What happens when `currentStep1` is the last step, but a new step is added after it? We need to invalidate the CP here, but without looking at the `length`, we don't know that something was added.

Reply:
CurrentStep always gets activated by the `scheduleOnce(this.updateFirstCurrentAndLastSteps)`

Length gets updated on add/remove so this means we end up getting the double modify error.

Reply:
HasNextStep was changing multiple times in a single  render due to length changing so much. CurrentStep changes only once per loop, and it usually does trigger a change, although I agree I don't know why.. I did write a test for this scenario. (In my usage, I add steps once I'm at the last step, so it seems to work)

Reply:
I'm not comfortable committing code that works if we don't know why it works.